kml/serdes_binary.hpp

```diff
@@ -253,19 +264,34 @@ class DeserializerKml
 
     if (m_header.m_version == Version::V8)
     {
-      // Check if file has Opensource V8 or MapsMe V8.
-      // Actual V8 format has 6 offset (uint64_t) in header. While V8MM has 5 offsets.
+      // Check if file has Opensource V8 or MapsMe V8 format.
+      // Actual V8 format has 6 offsets (uint64_t) in header. While V8MM has 5 offsets.
       // It means that first section (usually categories) has offset 0x28 = 40 = 5 * 8.
       if (m_header.m_categoryOffset == 0x28 || m_header.m_bookmarksOffset == 0x28 ||
           m_header.m_tracksOffset == 0x28 || m_header.m_stringsOffset == 0x28 ||
           m_header.m_compilationsOffset == 0x28)
       {
-        LOG(LWARNING, ("KMB file has version V8MM"));
+        LOG(LINFO, ("KMB file has version V8MM"));
         m_header.m_version = Version::V8MM;
         m_header.m_eosOffset = m_header.m_stringsOffset;
         m_header.m_stringsOffset = m_header.m_compilationsOffset;
       }
     }
+    if (m_header.m_version == Version::V9)
+    {
+      // Check if file has Opensource V9 or MapsMe V9 format.
+      // Actual V9 format has 6 offsets (uint64_t) in header. While V9MM has 5 offsets.
+      // It means that first section (usually categories) has offset 0x28 = 40 = 5 * 8.
+      if (m_header.m_categoryOffset == 0x28 || m_header.m_bookmarksOffset == 0x28 ||
+          m_header.m_tracksOffset == 0x28 || m_header.m_stringsOffset == 0x28 ||
+          m_header.m_compilationsOffset == 0x28)
+      {
+        LOG(LINFO, ("KMB file has version V9MM"));
+        m_header.m_version = Version::V9MM;
+        m_header.m_eosOffset = m_header.m_stringsOffset;
+        m_header.m_stringsOffset = m_header.m_compilationsOffset;
+      }
+    }
```

Comment:
This block is equivalent to the previous.
Why do we have all these copy-paste? With new TrackDataV9MM, FileDataV9MM etc?

Reply:
@vng can you please help and apply a cleaner patch?